april/palindrome1.c

```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+#include <string.h>
+
+int main()
+{
+    int t,i;
+    scanf("%d",&t);
+    for(i=0;i<t;i++)
+  {
+     char str[100];
+     gets(str);
+  }
+
+ {
+   char str1[20],str2[20];
+   int i,n, length = 0;
+   scanf("%[^\n]%*c",str1);
+
+   for (i = 0; str1[i] != '\0'; i++)
+   {
+      length++;
+   }
+
+   for (i = length - 1; i >= 0; i--)
+   {
+      str2[length - i - 1] = str1[i];
+   }
+
+   for (n = 1, i = 0; i < length; i++)
+   {
+      if (str2[i] != str1[i])
+      n = 0;
+   }
+
```

Comment:
uporer logic jodi nije nije chinta kore ber kore thakis, taile thik ase, no objection. Onno kothao theke dekhe jodi koris taile problem ase. 
 But ar che easy and efficient algorithm think kora jay. Tui jeita korsis seita thik ase , but not the best solution 
`char str;
scanf("%[^\n]%*c",str);
int i=0;
int l = strlen(str) -1; // kono string length bair korte strlen() function use kora hoy
int palicheck = 0;
while(i<l){
if(str[i] != str[l]){
palicheck = 1;
break;
}

if(palicheck == 1){
printf("not palindrome\n");
}else{
printf("palindrome\n");

}
}`